packages/expo-cli/src/credentials/views/SelectDistributionCert.ts

```diff
@@ -0,0 +1,96 @@
+import { IosDistCredentials, IosAppCredentials } from '../credentials';
+
+export type DistCertOptions = {
+  experienceName: string;
+  bundleIdentifier: string;
+  disableAutomode?: boolean;
+} & ISelectOptions<DistCert>;
+
+export class SelectDistributionCert implements ISelect<DistCert> {
+  async _choosePreferredCreds(
+    distCerts: IosDistCredentials[],
+    appCredentials: IosAppCredentials[],
+    experienceName: string
+  ) {
+    // prefer the one that matches our experienceName
+    const appCredentialsForExperience = appCredentials.filter(
+      cred => cred.experienceName === experienceName
+    );
+    for (let distCert of distCerts) {
+      if (appCredentialsForExperience.some(cred => cred.distCredentialsId === distCert.id)) {
+        return distCert;
+      }
+    }
+    // else choose an arbitrary one
+    return distCerts[0];
+  }
+
+  async select(ctx: Context, options: DistCertOptions): Promise<ISelect<DistCert> | DistCert> {
+    const backSelect = async (ctx: Context, options: DistCertOptions) =>
+      await this.select(ctx, options);
+    const newOptions = { ...options, disableAutomode: true, backSelect };
+
+    const expoUsername = ctx.user && ctx.user.username;
+    const existingCertificates = expoUsername
+      ? await iosDistView.getValidDistCerts(ctx.ios.credentials, ctx.appleCtx)
+      : [];
+    const choices = [];
+
+    if (existingCertificates.length > 0) {
+      choices.push({
+        name: '[Choose existing certificate] (Recommended)',
+        value: 'CHOOSE_EXISTING',
+      });
+      if (!options.disableAutomode) {
+        // autoselect creds if we find valid ones
+        const { experienceName } = options;
+        const autoselectedCertificate = await this._choosePreferredCreds(
+          existingCertificates,
+          ctx.ios.credentials.appCredentials,
+          experienceName
+        );
+        log(`Using Distribution Certificate: ${autoselectedCertificate.certId}`);
+        return autoselectedCertificate;
+      }
+    } else {
+      // If there aren't any existing certs, this is the only option available
+      return await this.handleAction(ctx, 'GENERATE', newOptions);
+    }
+
+    choices.push({ name: '[Add a new certificate]', value: 'GENERATE' });
+
+    const question: Question = {
+      type: 'list',
+      name: 'action',
+      message: 'Select an iOS distribution certificate to use for code signing:',
+      choices: choices,
+      pageSize: Infinity,
+    };
+
+    const { action } = await prompt(question);
+    return await this.handleAction(ctx, action, newOptions);
+  }
+
+  async handleAction(
+    ctx: Context,
+    action: string,
+    options: DistCertOptions
+  ): Promise<ISelect<DistCert> | DistCert> {
+    switch (action) {
+      case 'CHOOSE_EXISTING':
+        return await new iosDistView.UseExistingDistributionCert().select(ctx, options);
+      case 'GENERATE':
+        return await new iosDistView.CreateIosDist().select(ctx, options);
```

Comment:
Just to make sure. Are you aware that this will only create distribution certificate? It will be added to user credentials but won't be assigned to any experience

Reply:
Good catch :O I've changed this so that if there is a `bundleIdentifier` and `experienceName`, we assign it to an experience. 

I noticed that `credentials:manager` relies on this as well. With the new changes, if a user runs the manager and chooses to `create distribution certificate` in project context, it will assign it to an experience. Else it just creates the certificate. Lmk if that's ok!

Reply:
I would prefer to keep old behavior in `credentials:manager`, I agree that in most cases users would want to create and assign to experience, but I think actions there should be as basic as possible. Also, all the credentials are displayed in one place and what is connected to what should be clearly visible